lib/virt_autotest/common.pm

```diff
@@ -162,10 +162,45 @@ if (get_var("REGRESSION", '') =~ /xen/) {
             linuxrc => 'ifcfg="eth0=192.168.122.120/24,192.168.122.1,192.168.122.1"'
         }
     );
-
-    delete($guests{sles12sp4PV}) if (!is_sle('=12-SP4'));
-    delete($guests{sles12sp5HVM}) if (!is_sle('=12-SP5'));
-    delete($guests{sles15sp1HVM}) if (!is_sle('=15-SP1'));
+    # Filter out guests not allowed for the detected SLE version
+    if (is_sle('=12-SP5')) {
+        my @allowed_guests = qw(sles12sp5HVM sles12sp5PV sles15sp1HVM sles15sp1PV sles15sp5HVM sles15sp5PV);
+        foreach my $guest (keys %guests) {
+            delete $guests{$guest} unless grep { $_ eq $guest } @allowed_guests;
+        }
+    } elsif (is_sle('=12-SP4')) {
+        my @allowed_guests = qw(les12sp4HVM sles12sp4PV les12sp5HVM sles12sp5PV);
```

Comment:
Typo? les12sp4HVM -> sles12sp4HVM,  les12sp5HVM->sles12sp5HVM?

Reply:
yes, you are right, will update it.

Reply:
VR test also does not have these two VM:  sles12sp4HVM and sles12sp5HVM
http://openqa.qam.suse.cz/tests/56924#step/prepare_guests/26

Reply:
Rerun VR test after fixing.

Reply:
VR Test passed, include sles12sp4HVM sles12sp4PV sles12sp5HVM sles12sp5PV
http://openqa.qam.suse.cz/tests/56930#step/prepare_guests/26